### PR TITLE
use peer dependencies

### PR DESCRIPTION
**Pull request by christianvogt**
Would like some help with testing standalone planner.

### Mandatory
 - [x] What does this PR do? 
Move angular from `dependencies` to `peerDependencies`.

`npm link --production` must be used to omit installation of `devDependencies` when linking.

Updated build scripts where `npm link` was referenced.

See openshiftio/openshift.io#3686 for details as why we're moving to peer dependencies and the changes to webpack that impact npm link.

- [x] What issue/task does this PR references?
openshiftio/openshift.io#3686

- [ ] Are the tests Included?
No
___
### Optional
- [x] Is the documentation Included?
Updated docs to indicate usage of `npm link --production`

- [ ] Are the Release Notes included?

- [x] @mention(s) to expected reviewer(s) included
@michaelkleinhenz @sanbornsen

### CODE REVIEW

**Comment by christianvogt** [comment]
I ran the standalone planner with success.

I couldn't run the automated scripts though. I got errors that package.json could not be found. The path was completely wrong. 

I haven't tried the setup with docker.

File: package.json [hunk 1]
```diff
@@ -87,30 +100,26 @@
...
-    "patternfly-ng": "^3.1.3",
-    "reflect-metadata": "^0.1.10",
-    "rh-ngx-datatable": "12.0.16",
-    "rxjs": "^5.0.1",
-    "zone.js": "^0.8.18"
+    "rxjs": "^5.0.1"
   },
-  "peerDependencies": {},
   "devDependencies": {
+    "@angular/animations": "^4.4.0",
```
[the remark this answers is not in the document]
Comment (christianvogt) [reply package.json]: For every peer dependency, you also want them available as dev dependencies to run unit tests and other dev environments. But when another project depends on planner, the dev dependencies aren't installed.

**Comment by sudsen** [comment]
@christianvogt does it change any of our planner dev workflow commands?
For example, does it affect any of these following commands -
`npm run build -- --watch`, `npm run tests -- --unit` etc.

**Comment by christianvogt** [comment]
@sudsen those don't change. Only npm link changes to include the production flag.

**Comment by nimishamukherjee** [comment]
Could do:
- [x] npm link --production
- [x]  could run docker

**Pull request merged**